src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -105,40 +105,75 @@ public function remove(string $key)
      */
     public function getAlpha(string $key, string $default = ''): string
     {
-        return preg_replace('/[^[:alpha:]]/', '', $this->get($key, $default));
+        return preg_replace('/[^[:alpha:]]/', '', $this->getString($key, $default));
     }
 
     /**
      * Returns the alphabetic characters and digits of the parameter value.
      */
     public function getAlnum(string $key, string $default = ''): string
     {
-        return preg_replace('/[^[:alnum:]]/', '', $this->get($key, $default));
+        return preg_replace('/[^[:alnum:]]/', '', $this->getString($key, $default));
     }
 
     /**
      * Returns the digits of the parameter value.
      */
     public function getDigits(string $key, string $default = ''): string
     {
-        // we need to remove - and + because they're allowed in the filter
-        return str_replace(['-', '+'], '', $this->filter($key, $default, \FILTER_SANITIZE_NUMBER_INT));
+        return preg_replace('/[^[:digit:]]/', '', $this->getString($key, $default));
     }
 
     /**
      * Returns the parameter value converted to integer.
+     *
+     * @param bool $failOnInvalid Throw an exception when the value is not a valid int
      */
-    public function getInt(string $key, int $default = 0): int
+    public function getInt(string $key, int $default = 0, /*bool $failOnInvalid = false*/): int
     {
-        return (int) $this->get($key, $default);
+        $failOnInvalid = func_num_args() >= 3 ? func_get_arg(2) : false;
+
+        if (false === $failOnInvalid) {
```

Comment:
`if (!$failOnInvalid) {`

& no need to guard against non-bool values (aka let's remove the elseif below)